### PR TITLE
#168781696 -user can edit/delete comment

### PR DESCRIPTION
**Pull request by sabin18**
#### What does this PR do?
PR to allow user to edit or delete his comment
#### Description of Task to be completed?
- add editComment function
- add deleteComment function
- add comment table
#### How should this be manually tested?
- clone the repository : git clone https://github.com/andela/caret-bn-backend.git
- checkout to branch ft-edit-comment-168781696
- run sequelize db:migrate
- run sequelize db:seed:all
- use this route to edit comment `PUT :/api/v1/requests/comments/:id` id=>comment id.
- add comment in your body when you are editing comment.
- use this route to delete comment `DELETE: /api/v1/requests/comments/:id` id=>comment id.
#### Any background context you want to provide?
`N/A`
#### What are the relevant pivotal tracker stories?
[168781696](https://www.pivotaltracker.com/story/show/168781696)
#### Screenshots (if appropriate)
Screen Shot 2019-10-31 at 14 52 28
Screen Shot 2019-10-31 at 15 16 30

#### Questions:
`N/A`

### CODE REVIEW

**Review by Cheza-Dzabala — CHANGES_REQUESTED** [review]
Hello Roger,

This is what I've discovered. When I send an empty body of data I am returned with "Message Updated Successfully" please rectify this.

File: src/controllers/commentsController.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      if (!comments) {
+        return responseUtil(res, 404, strings.comments.error.COMMENT_NOT_FOUND);
+      }
+      if (comments.userId !== ownerId) {
+        return responseUtil(res, 403, strings.comments.error.NOT_OWNER);
+      }
+      models.comments.update({ comment, }, { where: { id, }, });
+      return responseUtil(res, 200, strings.comments.success.COMMENT_UPDATED);
+    });
+  }
```
Comment (nignanthomas) [inline src/controllers/commentsController.js]: your methods to `edit` and to `delete` are very much similar.
Kindly refactor it.
**Suggestion:** 
- Create a helper to find a comment, by passing the required parameters in the helper
- A helper to check if **not found** and **not owner**
- Then one to update by passing the required parameters in the helper.

File: src/controllers/commentsController.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }).then(comments => {
+      if (!comments) {
+        return responseUtil(res, 404, strings.comments.error.COMMENT_NOT_FOUND);
+      }
+      if (comments.userId !== ownerId) {
+        return responseUtil(res, 403, strings.comments.error.NOT_OWNER);
+      }
+      models.comments.update({ deleted: true, }, { where: { id, }, });
+      return responseUtil(res, 200, strings.comments.success.COMMENT_DELETED);
+    });
```
Comment (nignanthomas) [inline src/controllers/commentsController.js]: This the delete method I was talking about.
Kindly refactor.

**Review by nignanthomas — CHANGES_REQUESTED** [review]
Great job @sabin18 
Kindly implement the changes I requested and we are good

**Review by nignanthomas — APPROVED** [review]

**Review by Cheza-Dzabala — APPROVED** [review]

File: src/controllers/commentsController.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+const { COMMENT_UPDATED, COMMENT_DELETED } = strings.comments.success;
+
+export default class CommentsController {
+  static editComment(req, res) {
+    const { id } = req.params;
+    const { comment } = req.body;
+
+    commentsHelper.findComment(req).then(comments => {
+
```
Comment (nakiwuge) [inline src/controllers/commentsController.js]: remove this line

File: src/controllers/commentsController.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      commentsHelper.checkComment(req, comments, res, { comment }, id, COMMENT_UPDATED);
+
```
Comment (nakiwuge) [inline src/controllers/commentsController.js]: remove this line

File: src/controllers/commentsController.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    });
+  }
+
+  static deleteComment(req, res) {
+    const { id } = req.params;
+
+    commentsHelper.findComment(req).then(comments => {
+
```
Comment (nakiwuge) [inline src/controllers/commentsController.js]: remove this line

File: src/controllers/commentsController.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      commentsHelper.checkComment(req, comments, res, { deleted: true }, id, COMMENT_DELETED);
+
```
Comment (nakiwuge) [inline src/controllers/commentsController.js]: remove this line

File: src/helpers/commentsHelper.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+  if (!comments) {
+    return responseUtil(res, 404, strings.comments.error.COMMENT_NOT_FOUND);
+  }
+  if (comments.userId !== ownerId) {
+    return responseUtil(res, 403, strings.comments.error.NOT_OWNER);
+  }
+  models.comments.update(data, { where: { id, }, });
+  return responseUtil(res, 200, message);
+
```
Comment (nakiwuge) [inline src/helpers/commentsHelper.js]: remove this line

File: src/helpers/commentsHelper.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+const checkComment = (req, comments, res, data, id, message) => {
+  const ownerId = req.user.payload.id;
+
+  if (!comments) {
+    return responseUtil(res, 404, strings.comments.error.COMMENT_NOT_FOUND);
+  }
+  if (comments.userId !== ownerId) {
+    return responseUtil(res, 403, strings.comments.error.NOT_OWNER);
+  }
+  models.comments.update(data, { where: { id, }, });
```
Comment (nakiwuge) [inline src/helpers/commentsHelper.js]: add anew line

**Review by nakiwuge — CHANGES_REQUESTED** [review]
Good job @sabin18. Kindly work on the feedback below
- Return the updated comment object when a user successfully updates a comment
- Return the updated comment object when a user successfully deletes a comment
- Check the other comments I have left on your pr

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**